test/MultitokenCollector.test.ts

```diff
@@ -0,0 +1,409 @@
+    partners = await buildPartners(
+      [partner1, partner2, partner3, partner4],
+      PARTNER_SHARES
+    );
+
+    fakeERC20Tokens = await Promise.all(
+      Array(2)
+        .fill(await smock.fake<ERC20>('ERC20'))
+        .map((fakeToken: FakeContract<ERC20>) => {
+          fakeToken.transfer.returns(true);
+
+          return fakeToken;
+        })
+    );
+
+    collectorFactory = await smock.mock<MultitokenCollector__factory>(
+      'MultitokenCollector'
+    );
+  });
+
+  describe('constructor()', function () {
+    it('Should deploy a MultitokenCollector with single token', async function () {
+      partners = await buildPartners(
+        [partner1, partner2, partner3, partner4],
+        PARTNER_SHARES
+      );
+      const expectedTokens = fakeERC20Tokens.map(({ address }) => address)[0];
+      const expectedOwnerAddress = owner.address;
+
+      const collector: MockContract<MultitokenCollector> =
+        await collectorFactory.deploy(
+          expectedOwnerAddress,
+          [expectedTokens],
+          partners,
+          remainderDestination.address
+        );
+
+      const actualOwnerAddress = await collector.owner();
+      const actualTokens = await collector.getTokens();
+
+      expect(actualOwnerAddress, 'Failed to set the owner').to.equal(
+        expectedOwnerAddress
+      );
+      expect(actualTokens.toString(), 'Failed to set tokens').to.equal(
+        expectedTokens.toString()
+      );
+    });
+
+    it('Should deploy a MultitokenCollector with multiple tokens', async function () {
+      partners = await buildPartners(
+        [partner1, partner2, partner3, partner4],
+        PARTNER_SHARES
+      );
+      const expectedTokens = fakeERC20Tokens.map(({ address }) => address);
+      const expectedOwnerAddress = owner.address;
+
+      const collector: MockContract<MultitokenCollector> =
+        await collectorFactory.deploy(
+          owner.address,
+          expectedTokens,
+          partners,
+          remainderDestination.address
+        );
+      const actualOwnerAddress = await collector.owner();
+      const actualTokens = await collector.getTokens();
+
+      expect(actualOwnerAddress, 'Failed to set the owner').to.equal(
+        expectedOwnerAddress
+      );
+      expect(actualTokens.toString(), 'Failed to set tokens').to.equal(
+        expectedTokens.toString()
+      );
+    });
+
+    it('Should not let deploy with invalid shares', async function () {
+      const partners = await buildPartners(
+        [partner1, partner2, partner3, partner4],
+        [25, 25, 25, 26]
+      );
+
+      await expect(
+        collectorFactory.deploy(
+          owner.address,
+          fakeERC20Tokens.map(({ address }) => address),
+          partners,
+          remainderDestination.address
+        )
+      ).to.be.rejectedWith('Shares must add up to 100%');
+    });
+
+    it('Should not let deploy if the array of paMultitokenCollectorrtners is empty', async function () {
```

Comment:
Im not sure what are you trying to describe here.

Reply:
just accidentally pasted something there I believe. removing and also removing "let" for clatity